### DISCUSSION TITLE
Suspense is not supported on Bun

### DISCUSSION THREAD

**Discussion by rocascent** [remix-run/remix#9221] [Q&A]
For example, i use `redirect('/page')` or `<Link to='/page' />` to get to `/page`.

And i create a loader in `app/routes/page.tsx` which returns a `defer` with `Suspense` and `Await`.
```typescript
export async function loader({
  params,
}: LoaderFunctionArgs) {
  return defer({ info: new Promise(r => setTimeout(r, 2000)).then(() => 'ok') })
}
```
But it takes 2s to jump. When I use `<Link to='/page' reloadDocument/>`, the page jumps immediately and renders `Suspense`'s fallback component for two seconds. Why?
### ------------------
The reason for this problem seems to be that `<Suspense/>` is not supported on `Bun`.
Because this only occurs when I run `bun run --bun dev`. If I run `bun run dev`, the bug does not appear.

**Comment by kiliman** [comment]
Because Remix and React Router do not unmount your route components when you navigate, they simply fetch your loaders, update the context, and let React handle the rendering.

If you're rendering the same component tree, then `<Suspense>` will not remount, so it won't render the fallback component. You need to force React to remount using `<Suspense key>`. 

https://stackblitz.com/github/kiliman/remix-suspense?file=README.md

**Reply by kiliman** [reply]
`<Link reloadDocument>` forces a new document request (vs client side fetch). That's why it works differently.
